lib/tools/adb-commands.js

```diff
@@ -402,24 +408,34 @@ methods.isDataOn = async function () {
 /*
  * on: true (to turn on) or false (to turn off)
  */
-methods.setDataState = async function (on) {
-  await this.shell(['settings', 'put', 'global', 'mobile_data', on ? 1 : 0]);
+methods.setDataState = async function (on, isEmulator = false) {
+  if (isEmulator) {
+    await this.shell(['svc', 'data', on ? 'enable' : 'disable']);
+  } else {
+    await this.shell(['am', 'broadcast', '-a', `${SETTINGS_HELPER_ID}.data_connection`,
+                      '--es', 'setstatus', on ? 'enable' : 'disable']);
+  }
 };
 
 /*
  * opts: { wifi: true/false, data true/false } (true to turn on, false to turn off)
  */
-methods.setWifiAndData = async function ({wifi, data}) {
-  let wifiOpts = [],
-      dataOpts = [];
+methods.setWifiAndData = async function ({wifi, data}, isEmulator = false) {
   if (!_.isUndefined(wifi)) {
-    wifiOpts = ['-e', 'wifi', (wifi ? 'on' : 'off')];
+    this.setWifiState(wifi, isEmulator);
   }
   if (!_.isUndefined(data)) {
-    dataOpts = ['-e', 'data', (data ? 'on' : 'off')];
+    this.setDataState(data, isEmulator);
+  }
+};
+
+methods.setGeoLocation = async function (location, isEmulator = false) {
+  if (isEmulator) {
+    this.adbExec(['emu', 'geo', 'fix', ''+ location.longitude, '' + location.latitude]);
```

Comment:
@jlipps I've checked what people say about that and found the bug
https://code.google.com/p/android/issues/detail?id=206180
Is there any method in Node to set decimal splitter based on location?